datadog_checks_base/datadog_checks/base/utils/agent/utils.py

```diff
@@ -4,16 +4,16 @@
 
 
 def should_profile_memory(datadog_agent, check_name):
-    tracemalloc_whitelist = datadog_agent.get_config('tracemalloc_whitelist') or ''
-    if tracemalloc_whitelist:
-        tracemalloc_whitelist = [check.strip() for check in tracemalloc_whitelist.split(',')]
-        tracemalloc_whitelist = [check for check in tracemalloc_whitelist if check]
+    tracemalloc_include = datadog_agent.get_config('tracemalloc_include') or datadog_agent.get_config('tracemalloc_whitelist') or ''
```

Comment:
What if both are set ? Could happen if the old was was set through env var and the new one with the datadog.yaml (very likely if customers update one and not the other).

We might want to check if both are set and different. If so, print a warning and use the new `tracemalloc_include`. WDYT ?

Reply:
Strictly speaking it would be better, but I though it's okay not to check specially for a debug feature like this... but given you found it missing, I'll add it.

Reply:
Also I'm not sure which logger we can always guarantee will be available from this part of the code. Do you know?

Reply:
We could check on the Go side if the logger is not ready. The pythong->go logger is available as soon as we load RTLoader and will be available here. If we don't have a check we might have to use the builtin directly (`datadog_agent.log(message, log_level)`).

I think it's important to check, as it would be very confusing to an SRE/admin-sys to not have his env var being used because the configuration shipped in the container/image/... use the new naming.